blueocean-config/src/main/java/io/jenkins/blueocean/config/BlueOceanConfigFactory.java

```diff
@@ -0,0 +1,62 @@
+package io.jenkins.blueocean.config;
+
+import javax.annotation.Nonnull;
+
+import hudson.ExtensionPoint;
+import jenkins.model.Jenkins;
+
+/**
+ * Blue Ocean configuration factory which allows different extensions to override the default behavior.
+ * 
+ */
+public abstract class BlueOceanConfigFactory implements ExtensionPoint {
+    public static final String FEATURE_PROPERTY_PREFIX = "blueocean.features.";
+    public static final String ORGANIZATIONS_ENABLED = "organizations.enabled";
+
+    private static BlueOceanConfig config;
```

Comment:
Looks like intent is to cache config instance, why not use cache such as guava? Or maybe use read/write lock instead of synchronization block on the method?